files/common/lib/systemd/system/systemd-reexec.service

```diff
@@ -0,0 +1,26 @@
+#
+# Copyright 2023 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+[Unit]
```

Comment:
Let's add a comment here explaining why this service exists. Someone is bound to see the service getting run on a live engine in production and want to know why it's there by cat'ing the unit file.

Reply:
sure.. I put some comments in https://github.com/delphix/appliance-build/pull/713 .. but I think it makes sense to add some here too..

Reply:
done.. added some comments.